tests/aws/services/sqs/test_sqs_move_task.py

```diff
@@ -0,0 +1,452 @@
+
+    # receive each message two times to move them into the dlq
+    for i in range(n * 2):
+        sqs.receive_message(QueueUrl=queue_url, VisibilityTimeout=0)
+
+    # wait until the messages arrive in the DLQ
+    assert sqs_wait_queue_size(sqs, dl_queue_url, expected_num_messages=n, timeout=10) == n
+
+    # start move task
+    response = aws_client.sqs.start_message_move_task(
+        SourceArn=source_arn, DestinationArn=destination_arn, MaxNumberOfMessagesPerSecond=1
+    )
+    snapshot.match("start-message-move-task-response", response)
+    started = time.time()
+    messages = sqs_collect_messages(sqs, destination_queue, n, 60)
+    assert {message["Body"] for message in messages} == {
+        "message-0",
+        "message-1",
+        "message-2",
+        "message-3",
+    }
+
+    assert time.time() - started >= 3
+
+
+@markers.aws.validated
+@pytest.mark.skip_snapshot_verify(
+    paths=[
+        # this is non-deterministic because of concurrency in AWS vs LocalStack
+        "$..ApproximateNumberOfMessagesMoved",
+    ]
+)
+def test_move_task_cancel(
+    sqs_create_queue,
+    sqs_create_dlq_pipe,
+    sqs_get_queue_arn,
+    aws_client,
+    snapshot,
+):
+    sqs = aws_client.sqs
+
+    # create dlq pipe: some-queue -> dlq (source) -> destination
+    queue_url, dl_queue_url = sqs_create_dlq_pipe(max_receive_count=1)
+    source_arn = sqs_get_queue_arn(dl_queue_url)
+    destination_queue = sqs_create_queue()
+    destination_arn = sqs_get_queue_arn(destination_queue)
+
+    n = 10
+
+    # send n messages
+    for i in range(n):
+        sqs.send_message(QueueUrl=queue_url, MessageBody=f"message-{i}")
+
+    assert sqs_wait_queue_size(sqs, queue_url, expected_num_messages=n, timeout=10) == n
+
+    # receive each message two times to move them into the dlq
+    for i in range(n * 2):
+        sqs.receive_message(QueueUrl=queue_url, VisibilityTimeout=0)
+
+    # wait until the messages arrive in the DLQ
+    assert sqs_wait_queue_size(sqs, dl_queue_url, expected_num_messages=n, timeout=10) == n
+
+    # start move task
+    response = aws_client.sqs.start_message_move_task(
+        SourceArn=source_arn, DestinationArn=destination_arn, MaxNumberOfMessagesPerSecond=1
+    )
+    task_handle = response["TaskHandle"]
+
+    # wait for two messages to arrive, then cancel the task
+    messages = sqs_collect_messages(sqs, destination_queue, 2, 60)
+    assert len(messages) == 2
+
+    response = sqs.list_message_move_tasks(SourceArn=source_arn)
+    snapshot.match("list-while", response)
+
+    response = sqs.cancel_message_move_task(TaskHandle=task_handle)
+    snapshot.match("cancel", response)
+
+    # check move task completion (in AWS, approximate number of messages may take a while to update)
+    def _wait_for_task_cancellation():
+        _response = aws_client.sqs.list_message_move_tasks(SourceArn=source_arn)
+        assert _response["Results"][0]["Status"] == "CANCELLED"
+        return _response
```

Comment:
Here we check for the status, but not the approx num which is mentioned in the comments (which we checked in the in the test commented on before).

Reply:
i just wanted at least one test that covers that `ApproximateNumberOfMessagesMoved` is set correctly eventually some point. as noted in a comment, in AWS, approximate number of messages may take a while to update, even after the "Status" field has been set to COMPLETED. :shrug: